src/Core/DecimalFunctions.h

```diff
@@ -183,7 +183,10 @@ To convertTo(const DecimalType & decimal, size_t scale)
 
     if constexpr (std::is_floating_point_v<To>)
     {
-        return static_cast<To>(decimal.value) / static_cast<To>(scaleMultiplier<NativeT>(scale));
+        if constexpr (is_big_int_v<NativeT>)
+            return static_cast<To>(decimal.value) / static_cast<To>(scaleMultiplier<NativeT>(scale));
+        else
+            return static_cast<To>(decimal.value) / scaleMultiplier<NativeT>(scale);
```

Comment:
Is this the significant part? Why does it improve things, is `scaleMultiplier` for big ints some large type? Some comment would help.

Reply:
There're lots of such places for big ints over the code, it's not possible to comment all of them. It's better to remove boost::multiprecision ints with betterones one day. boost::multiprecision has no implicit casts for its types, so we have to make explicit `static_cast<To>()` for them. But nested types could have special mix-typed operations. Here we divide Float64 over NativeT that could be faster then static_cast<Float64>(T) + Float64 over Float64. We had perf degradation for Int64 and Int128 nested types and no degradation for Int32 in decimal aggregates with additional cast to Float here.

Reply:
OK, I see, let's wait for the test. About comments -- it's true that we can't comment everything, but you still had to explain this part to me here. Personally I prefer just to put these explanations into the code, so that they are easily accessible to the future readers as well.

```suggestion
        // boost::multipreision has no implicit casts, so we have to explicitly cast it to Float here. On the other hand, native
        // types might have more efficient implementations for division of Float by NativeT, so it might be more efficient to
        // avoid the cast.
        if constexpr (is_big_int_v<NativeT>)
            return static_cast<To>(decimal.value) / static_cast<To>(scaleMultiplier<NativeT>(scale));
        else
            return static_cast<To>(decimal.value) / scaleMultiplier<NativeT>(scale);
```